### DISCUSSION TITLE
XYZ Offset to compensate Dial Indicator position in Manual Mesh levelling

### DISCUSSION THREAD

**Discussion by jetel73** [Jyers/Marlin#292] [Ideas]
Hello Jacob,

I'll probably never use any Bl Touch or so on this machine. Essentially because I don't want to modify my printer, and because I love your Manual Mesh Levelling implementation. It is just the option I was trying to learn Marlin for when I discovered your work.

As I previously asked on your Reddit Thread, would it be possible, in the future, to add an offset menu to allow the use of any type of comparator or dial indicator attached to the head when doing the manual mesh levelling procedure.

I'm quite sure that people that use a comparator to level their bed would like this possibility very much. It would be a lot faster, to correct on the go each point of the mesh.

For example, an "Adjust comparator position" or so option in the manual mesh levelling menu, to set the X Y Z position of the dial indicator and store it, before the beginning of the procedure. Next time, only Z position to adjust...

Anyway your firmware is really pleasant to use, I like the new numeration 1.2.1 - 1.2.2 etc... It's perfectly clear that way and really easy to follow the progresses. 

Very good job
Regards
Jetel

**Comment by Cal-LF** [comment]
I use the dial indicator for initial leveling, which is quick to do. After that, just live adjustments are needed. Would manual mesh leveling work with a dial indicator now? It would be the same as using paper to set the gap.

**Reply by jetel73** [reply]
There is no possibility, no menu in the 1.2.2 Manual mesh levelling version to set the X Y offset as it is in the Bltouch version.

I use this dial indicator (https://www.thingiverse.com/thing:4524389) the reading is direct and precise. For me a paper sheet can in no way be as reliable and precise as a direct reading.

Do you use the Manual Mesh levelling function or a BlTouch ?

**Reply by Cal-LF** [reply]
I have not used any manual mesh leveling yet. Thought it could be a possibility to use the dial indicator with MM. After final manual leveling with the dial indicator, I adjust the wheels while printing the first layer. After that the bed stays aligned.
My bed is not warped yet so manual mesh is not needed. This is the holder I use. https://www.thingiverse.com/thing:4587559

**Reply by jetel73** [reply]
Believe me, as soon as you try the mesh levelling function you'll adopt it.
I never touch my wheels in any manner during a print and I hate to think having to do so. My manual level is as accurate as possible and the mesh does the local corrections to have a perfect first layer. 
I have implemented the M420 S1 Z2 in the beginning Gcode. The compensation is sprayed over the following layers until my print reaches the height of 2mm as the Z2 command leads the printer to.
Try it !

**Comment by Jyers** [comment]
I have been meaning to test this and will definitely put it on my todo-list. It just depends on whether probe offsets are still applied when probe manually is enabled.

**Reply by jetel73** [reply]
Very glad to read this thx !

**Reply by Cal-LF** [reply]
That would be a very nice addition to use a visual probe like the dial indicator.
